src/main/java/ch/njol/skript/lang/Section.java

```diff
@@ -66,6 +91,47 @@ protected void loadCode(SectionNode sectionNode) {
 		}
 	}
 
+	/**
+	 * Loads the code in the given {@link SectionNode},
+	 * appropriately modifying {@link ParserInstance#getCurrentSections()}.
+	 *
+	 * This method differs from {@link #loadCode(SectionNode)} in that it
+	 * is meant for code that will be executed in a different event.
+	 *
+	 * @param sectionNode The section node to load.
+	 * @param name The name of the event(s) being used.
+	 * @param events The event(s) during the section's execution.
+	 * @return A trigger containing the loaded section. This should be stored and used
+	 * to run the section one or more times.
+	 */
+	@SafeVarargs
+	protected final Trigger loadCode(SectionNode sectionNode, String name, Class<? extends Event>... events) {
+		ParserInstance parser = getParser();
+
+		String previousName = parser.getCurrentEventName();
+		Class<? extends Event>[] previousEvents = parser.getCurrentEvents();
+		SkriptEvent previousSkriptEvent = parser.getCurrentSkriptEvent();
+		List<TriggerSection> previousSections = parser.getCurrentSections();
+		Kleenean previousDelay = parser.getHasDelayBefore();
+
+		parser.setCurrentEvent(name, events);
+		parser.setCurrentSkriptEvent(null);
```

Comment:
We could set this to a FakeSkriptEvent (same one as used for creating the trigger object)